webpack/common.js

```diff
@@ -0,0 +1,290 @@
+	'googlesitekit-data': [ 'googlesitekit', 'data' ],
+	'googlesitekit-modules': [ 'googlesitekit', 'modules' ],
+	'googlesitekit-widgets': [ 'googlesitekit', 'widgets' ],
+	'googlesitekit-components': [ 'googlesitekit', 'components' ],
+	'@wordpress/i18n': [ 'googlesitekit', 'i18n' ],
+};
+
+const externals = { ...siteKitExternals };
+
+const noAMDParserRule = { parser: { amd: false } };
+
+const svgRule = {
+	test: /\.svg$/,
+	use: [
+		{
+			loader: '@svgr/webpack',
+			options: {
+				// strip width & height to allow manual override using props
+				dimensions: false,
+			},
+		},
+	],
+};
+
+const createRules = ( mode ) => [
+	noAMDParserRule,
+	svgRule,
+	{
+		test: /\.js$/,
+		exclude: /node_modules/,
+		use: [
+			{
+				loader: 'babel-loader',
+				options: {
+					sourceMap: mode !== 'production',
+					babelrc: false,
+					configFile: false,
+					cacheDirectory: true,
+					presets: [ '@wordpress/default', '@babel/preset-react' ],
+				},
+			},
+		],
+		...noAMDParserRule,
+	},
+	{
+		test: RegExp( 'node_modules/@material/web/.*.js' ),
+		use: [
+			{
+				loader: 'babel-loader',
+				options: {
+					sourceMap: mode !== 'production',
+					babelrc: false,
+					configFile: false,
+					cacheDirectory: true,
+					presets: [ '@wordpress/default', '@babel/preset-react' ],
+				},
+			},
+		],
+		...noAMDParserRule,
+	},
+];
+
+const createMinimizerRules = ( mode ) => [
+	new TerserPlugin( {
+		parallel: true,
+		sourceMap: mode !== 'production',
+		cache: true,
+		terserOptions: {
+			// We preserve function names that start with capital letters as
+			// they're _likely_ component names, and these are useful to have
+			// in tracebacks and error messages.
+			keep_fnames: /__|_x|_n|_nx|sprintf|^[A-Z].+$/,
+			output: {
+				comments: /translators:/i,
+			},
+		},
+		extractComments: false,
+	} ),
+];
+
+// Get the app version from the google-site-kit.php file - optional chaining operator not supported here
+const googleSiteKitFile = fs.readFileSync(
+	path.resolve( rootDir, 'google-site-kit.php' ),
+	'utf8'
+);
+const googleSiteKitVersion = googleSiteKitFile.match(
+	/(?<='GOOGLESITEKIT_VERSION',\s+')\d+.\d+.\d+(?=')/gi
+);
+const GOOGLESITEKIT_VERSION = googleSiteKitVersion
+	? googleSiteKitVersion[ 0 ]
+	: '';
+
+const corePackages = [
+	'api-fetch',
+	'compose',
+	'data',
+	'dom-ready',
+	'element',
+	'icons',
+	'keycodes',
+	'url',
+];
+
+const gutenbergExternals = {
+	'@wordpress/i18n': [ 'googlesitekit', 'i18n' ],
+};
+
+corePackages.forEach( ( name ) => {
+	gutenbergExternals[ `@wordpress-core/${ name }` ] = [
+		'wp',
+		name.replace( /-([a-z])/g, ( match, letter ) => letter.toUpperCase() ),
+	];
+} );
+
+module.exports = {
+	rootDir,
+	formattedFeaturesToPHPArray,
+	resolve,
+	manifestTemplate,
+	configTemplate,
+	manifestArgs,
+	siteKitExternals,
+	externals,
+	gutenbergExternals,
+	noAMDParserRule,
+	svgRule,
+	createMinimizerRules,
+	createRules,
+	GOOGLESITEKIT_VERSION,
+	corePackages,
+	projectPath,
+};
```

Comment:
Instead of having this exports statement, it is better to define exports where it is declared:

```diff
- const GOOGLESITEKIT_VERSION = googleSiteKitVersion
+ exports.GOOGLESITEKIT_VERSION = googleSiteKitVersion
  	? googleSiteKitVersion[ 0 ]
  	: '';

- const gutenbergExternals = {
+ exports.gutenbergExternals = {
  	'@wordpress/i18n': [ 'googlesitekit', 'i18n' ],
  };

  ...
```
In this case, we don't need to keep in mind that we need to update the export object at the bottom if we need to add something new to this file.